addons/stock/models/stock_move_line.py

```diff
@@ -219,7 +224,14 @@ def _onchange_qty_done(self):
     def _get_default_dest_location(self):
         if self.env.context.get('default_location_dest_id'):
             return self.env['stock.location'].browse([self.env.context.get('default_location_dest_id')])
-        return self.move_id.location_dest_id or self.picking_id.location_dest_id or self.location_dest_id
+        return (self.move_id.location_dest_id or self.picking_id.location_dest_id or self.location_dest_id)[0]
+
+    def _get_putaway_additional_qty(self):
+        addtional_qty = {}
+        for ml in self._origin:
+            qty = max(ml.product_uom_id._compute_quantity(ml.qty_done, ml.product_id.uom_id), ml.product_uom_qty)
+            addtional_qty[ml.location_dest_id.id] = addtional_qty.get(ml.location_dest_id.id, 0) - qty
+        return addtional_qty
```

Comment:
IMO this should be compute in `_get_putaway_strategy`

Reply:
If so, we maybe want to move `_get_putaway_strategy` from `stock.location` to `stock.move`. Maybe better do it only in master?